app/models/concerns/foreman_bootdisk/compute_resources/proxmox.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module ForemanBootdisk
+  module ComputeResources
+    module Proxmox
+      def capabilities
+        super + [:bootdisk]
+      end
+
+      def iso_upload(iso, vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+        server.ssh_options = { password: fog_credentials[:proxmox_password] }
+        server.ssh_ip_address = server.node_id
+        server.username = client.credentials[:current_user].split('@').first
+        server.scp_upload(iso, '/var/lib/vz/template/iso/')
+        server.reload
+        storage = storages(server.node_id, 'iso')[0]
+        storage.volumes.any? { |v| v.volid.include? File.basename(iso) }
+      end
+
+      def iso_attach(iso, vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+        storage = storages(server.node_id, 'iso')[0]
+        volume = storage.volumes.detect { |v| v.volid.include? File.basename(iso) }
+        disks = server.disks.map { |disk| disk.split(":")[0] }.join(";")
+        config_hash = { ide2: "#{volume.volid},media=cdrom" }
+        server.update(config_hash)
+        server.update({ boot: "order=ide2;#{disks}" })
```

Comment:
AFAIK the latest rubocop-performance will flag this and tell you to simplify to regular assignments:
```suggestion
        server[:ide2] = "#{volume.volid},media=cdrom"
        server[:boot] = "order=ide2;#{disks}"
```

Reply:
Since it's a server object and not a hash so this assignment is not possible and it throw error NoMethodError (undefined method []=' for #Fog::Proxmox::Compute::Server:0x0000564003398f20)`

Reply:
And we want to first add the cdrom (ide2) and update the server and then update the boot order because if we update the server with both cdrom and boot order, it takes wrong boot order. It tries to update the attributes alphabetically and without adding `ide2` it cannot update boot order properly.